pkg/ratelimit/concurrency_limiter.go

```diff
@@ -26,11 +26,13 @@ func newConcurrencyLimiter(limit uint64) *concurrencyLimiter {
 	return &concurrencyLimiter{limit: limit}
 }
 
+const unlimit = uint64(0)
+
 func (l *concurrencyLimiter) allow() bool {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
-	if l.current+1 <= l.limit {
+	if l.limit == unlimit || l.current+1 <= l.limit {
```

Comment:
Why do we need it?

Reply:
Reusing `concurrencyLimiter` statistics, and add concurrencyLimiter to all APIs. So now when limit is equal to 0, there is no limit

Reply:
if l.limit is equal to 0, this condition will always be false?

Reply:
if l.limit is equal to 0, we should always allow